### PR TITLE
[Improve] Beautify the YOLOX configuration

### PR DESCRIPTION
**Pull request by lyviva**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation for this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a potential influence on downstream projects, this PR should be tested with downstream projects, like MMDetection or MMClassification.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: configs/yolox/yolox_s_fast_8xb8-300e_coco.py [hunk 1]
```diff
@@ -1,21 +1,66 @@
...
+widen_factor = 0.5
+norm_cfg = dict(type='BN', momentum=0.03, eps=0.001)
+
+# -----train val related-----
+weight_decay = 0.0005
 num_last_epochs = 15
+affine_scale_left = 0.1
+affine_scale_right = 2
+Mixup_scale_left = 0.8
+Mixup_scale_right = 1.6
```
Comment (PeterH0323) [inline configs/yolox/yolox_s_fast_8xb8-300e_coco.py:57]:
```suggestion
mixup_scale_left = 0.8
mixup_scale_right = 1.6
```

Comment (PeterH0323) [reply configs/yolox/yolox_s_fast_8xb8-300e_coco.py:57]: 变量名要小写

File: configs/yolox/yolox_s_fast_8xb8-300e_coco.py [hunk 2]
```diff
@@ -1,21 +1,66 @@
...
+train_data_prefix = 'train2017/'  # Prefix of train image path
+# path of val annotation file
+val_ann_file = 'annotations/instances_val2017.json'
+val_data_prefix = 'val2017/'  # Prefix of train image path
 
-save_epoch_intervals = 10
-train_batch_size_per_gpu = 8
+num_classes = 80  # Number of classes for classification
+# Batch size of a single GPU during training
+train_batch_size_per_gpu = 16
```
Comment (hhaAndroid) [inline configs/yolox/yolox_s_fast_8xb8-300e_coco.py:15]:
```suggestion
train_batch_size_per_gpu = 8
```

File: configs/yolox/yolox_s_fast_8xb8-300e_coco.py [hunk 3]
```diff
@@ -113,13 +153,13 @@
         pre_transform=pre_transform),
     dict(
         type='mmdet.RandomAffine',
-        scaling_ratio_range=(0.1, 2),
+        scaling_ratio_range=(affine_scale_left, affine_scale_right),
```
Comment (hhaAndroid) [inline configs/yolox/yolox_s_fast_8xb8-300e_coco.py:156]: 不用这么写，你要不直接把 scaling_ratio_range 放到前面去，不要分啥left right

File: configs/yolox/yolox_s_fast_8xb8-300e_coco.py [hunk 4]
```diff
@@ -113,13 +153,13 @@
...
         # img_scale is (width, height)
         border=(-img_scale[0] // 2, -img_scale[1] // 2)),
     dict(
         type='YOLOXMixUp',
         img_scale=img_scale,
-        ratio_range=(0.8, 1.6),
+        ratio_range=(mixup_scale_left, mixup_scale_right),
```
Comment (hhaAndroid) [inline configs/yolox/yolox_s_fast_8xb8-300e_coco.py:162]: 这个也一样修改

File: configs/yolox/yolox_s_fast_8xb8-300e_coco.py [hunk 5]
```diff
@@ -201,19 +241,22 @@
...
     type='OptimWrapper',
     optimizer=dict(
-        type='SGD', lr=base_lr, momentum=0.9, weight_decay=5e-4,
-        nesterov=True),
+        type='SGD',
+        lr=base_lr,
+        momentum=0.9,
+        weight_decay=weight_decay,
+        nesterov=True,
+        train_batch_size_per_gpu=train_batch_size_per_gpu),
```
Comment (hhaAndroid) [inline configs/yolox/yolox_s_fast_8xb8-300e_coco.py:259]: 没有这个参数

File: configs/yolox/yolox_tiny_fast_8xb8-300e_coco.py [hunk 6]
```diff
@@ -1,8 +1,15 @@
 _base_ = './yolox_s_fast_8xb8-300e_coco.py'
 
+# ========================modified parameters======================
 deepen_factor = 0.33
 widen_factor = 0.375
 
+img_scale = _base_.img_scale
+pre_transform = _base_.pre_transform
+scaling_ratio_range = (0.5, 1.5)
+random_flip_prob = 0.5
```
Comment (hhaAndroid) [inline configs/yolox/yolox_tiny_fast_8xb8-300e_coco.py:10]: 不用放前面，大家不会改

File: configs/yolox/yolox_tiny_fast_8xb8-300e_coco.py [hunk 7]
```diff
@@ -1,24 +1,30 @@
...
 model = dict(
-    data_preprocessor=dict(batch_augments=[
-        dict(
-            type='mmdet.BatchSyncRandomResize',
-            random_size_range=(320, 640),  # note
-            size_divisor=32,
-            interval=10)
-    ]),
+    data_preprocessor=dict(
+        type='YOLOv5DetDataPreprocessor',
```
Comment (hhaAndroid) [inline configs/yolox/yolox_tiny_fast_8xb8-300e_coco.py:15]: 这里不用改

**Review by hhaAndroid — APPROVED** [review]

**Pull request merged**